src/index.ts

```diff
@@ -34,8 +29,14 @@ namespace Miniget {
     backoff: { inc: number; max: number };
   }
 
+  export interface HTTPLib {
+    request: (options: RequestOptions | string | URL,
+    callback?: (res: IncomingMessage) => void) => ClientRequest;
+  }
+
   export type defaultOptions = Miniget.Options;
   export type MinigetError = Error;
+  export type httpLibs = { [key: string]: HTTPLib };
```

Comment:
this is exported for the tests right? look at https://github.com/fent/node-muk-require

it would work something like

```ts
const badMiniget = muk('../', {
  http: null,
  https: null,
});
```

Reply:
i know about mocking - have only used proxies for it though and that was just awful...
only problem is that library is not ts campatible😉

Reply:
i'll ask the author if they wanna convert to ts